### DISCUSSION TITLE
Liebherr 42146 with one remote

### DISCUSSION THREAD

**Discussion by Dennis1425** [pybricks/support#1447] [Show and tell]
**Setup:**
Put the program on both hubs.
Upon starting the program the hub will search for a remote for 5 seconds, after that it will assume the other hub has the remote, so it will go into bluetooth receiving state.

So turn on both hubs, turn on the remote, one hub will connect to the remote, the other will receive commands over bluetooth, it doesnt matter which hub has the remote.

**Files:**

[lego_42146_pybricks.zip](https://github.com/pybricks/support/files/14311940/lego_42146_pybricks.zip)
- L42146_main.py
Is the main program entry point, figures out which hub it is based on the motor connection, if you have your motors differently you might need to adjust this.
- L42146_definitions.py
Has all the shared code, also a few globals if you want to adjust maximum speed of the motors or
 the bluetooth channel used.

**Controls:**

- **Remote green light**
Controlling the upper hub
  - **left/right plus**
hoist
  - **left/right middle buttons**
jib
  - **left/right minus buttons**
boom
  - **center Button**
switch to lower hub

- **Remote blue light**
Controlling the lower hub
  - **Left plus/minus**
left track
  - **right plus/minus**
right track
  - **left/right middle buttons**
Rotation
  - **center Button**
switch to upper hub

**Comment by laurensvalk** [comment]
Awesome, thanks for sharing!

**Comment by BertLindeman** [comment]
Funny.

My programs.
Free for non commercial use though.

Bert

**Comment by vanczakp** [comment]
Hi! Checked your code, please accept some remarks:
-top speed of the angular motor is around 1150deg/sec without any load. You can check it with a simple program. 1500 cannot be reached,
-controller's analog stick almost never goes back to 0. So you always between 0-6. Filter your paramter and you wont hear again the high pitch noise of the motors.

Anyway I almost finished my variant, with one control, everything is analogue controlled, different control optinons, speed options and calibration to have end positions of booms and hoist. Weight measurment/overload warning will be only available when vibration and or color change of controller is possible.
Both analogue stick are is converted to square so at 45deg to top right gives x=100 and y=100 isntead of 71 and 71. Track controls like a tank with one stick.
Will upload a video later.

**Reply by BertLindeman** [reply]
Good you look into this 😃 

> Hi! Checked your code, please accept some remarks: 
-top speed of the angular motor is around 1150deg/sec without any load. You can check it with a simple program. 1500 cannot be reached, 

Oke, but why would that be important?

> -controller's analog stick almost never goes back to 0. So you always between 0-6. Filter your paramter and you wont hear again the high pitch noise of the motors.

This is already built-in in pybricks see #1473  
So there is no need to filter in all our python programs.
